### PR TITLE
Enhancements in Mission module

### LINKED ISSUE #404: Enhancement in mission segments
Some changes would be needed in mission segments to prepare future evolutions, e.g. a takeoff segment:
- FlightSegment._get_gamma_and_acceleration() should take a FlightPoint as input, to allow the calculation to depend on more parameters than mass, drag and thrust
- dISA should be settable
- The definition of a starting point (especially altitude) should be possible
- It could be useful to have the possibility of defining in Python a segment that is a combination of other segments.

### PR DESCRIPTION
**Pull request by christophe-david**
Closes #404.

- Documentation has been updated.  Preview [here](https://fast-oad.readthedocs.io/en/mission-fixes/documentation/mission_module/index.html)
  - Description of mission definition and segments has been update
  - A section has been added about extensibility. Description of FlightPoint class has been moved in there.
- `FlightSegment._get_gamma_and_acceleration()` now take a FlightPoint as input, to allow the calculation to depend on more parameters than mass, drag and thrust
- a parameter with a variable as value can now be associated to a unit and a default value that will be used in the OpenMDAO input declaration (and be in generated input data file)
- a target parameter can be declared as relative to the start point of the segment by prefixing the parameter name with `delta_`
- when setting a parameter, a minus sign can be put before a variable name to get the opposite value (can be useful with relative values)
- a parameter can now be set at route or mission level
- dISA can now be set in mission definition file with `isa_offset`
- taxi-out and takeoff are now set in mission definition file:
  - mission starting point (altitude, speed, mass) can now be set using the "start" segment
  - the mass input of the mission can be set using the "mass input" segment. This segment can be anywhere in the mission, though it is expected that fuel consumption in previous segments is mass-independent.
- a mission phase can now contain other phases 
- if a segment parameter (dataclass field) is an array or a list, the associated variable in mission file will be declared with `shape_by_conn=True`

### CODE REVIEW

File: docs/documentation/mission_module/extensibility/adding_segments.rst [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+The main field of the class will be
+:attr:`~fastoad.models.performances.mission.segments.base.AbstractFlightSegment.target`,
+provided as a :ref:`FlightPoint instance <flight-point>`, which
+will contain the flight point parameters set as target in the mission definition file.
+
+The instantiation in FAST-OAD will be like this::
+
+    import fastoad.api as oad
+
+    segment = NewSegment( target=oad.FlightPoint(altitude=5000.0, true_airspeed=200.0,
```
Comment (esnguyenvan) [inline docs/documentation/mission_module/extensibility/adding_segments.rst:156]: There is a parenthesis missing here

File: docs/documentation/mission_module/mission_file/setting_values.rst [hunk 2]
```diff
@@ -164,11 +172,78 @@ Example 2 : basic contextual variable in a flight phase
...
+    * - 1
+      - \data:mission
+      - mission_1:route_A
+      - distance
+      - \data:mission:mission_1:route_A:distance
+    * - 1
+      - \data:mission
+      - mission_2:route_A
+      - distance
+      - \data:mission:mission_1:route_A:distance
```
Comment (esnguyenvan) [inline docs/documentation/mission_module/mission_file/setting_values.rst:203]:
```suggestion
      - \data:mission:mission_2:route_A:distance
```

File: docs/documentation/mission_module/mission_file/setting_values.rst [hunk 3]
```diff
@@ -164,11 +172,78 @@ Example 2 : basic contextual variable in a flight phase
...
+
+
+**Examples 2 & 3**
+
+:code:`phase_a` contains the parameters :code:`thrust_rate` and :code:`time_step` where contextual
+variable names are affected.
+For :code:`thrust_rate`, default prefix (:code:`data:mission:`) and suffix (:code:`thrust_rate`)
+will be used.
+For :code:`time_step`, prefix is customized (:code:`settings:mission`) and default prefix
```
Comment (esnguyenvan) [inline docs/documentation/mission_module/mission_file/setting_values.rst:212]:
```suggestion
For :code:`time_step`, prefix is customized (:code:`settings:mission`) and default suffix
```

**Review by esnguyenvan — APPROVED** [review]
I mainly propose modifications in the documentation. I have practised the new code in branch 'mission-takeoff', I find it better organized than last version and the associated documentation should ease the addition of custom segments which was on of the goal. Overall it should allow more flexible mission definition and usage.

File: docs/documentation/mission_module/mission_file/mission_file.rst [hunk 4]
```diff
@@ -155,9 +174,8 @@ Example:
 
 .. _mission-section:
 
-***************
-Mission section
-***************
+Mission definition section
```
[the remark this answers is not in the document]
Comment (christophe-david) [reply docs/documentation/mission_module/mission_file/mission_file.rst:177]: Fixed in 7ee9fefe6b0d269e29479d2e03e24b4a0a3b9cc4

**Pull request merged**